app/javascript/mastodon/components/status_action_bar.js

```diff
@@ -360,8 +360,8 @@ class StatusActionBar extends ImmutablePureComponent {
     return (
       <div className='status__action-bar'>
         <IconButton className='status__action-bar__button' title={replyTitle} icon={status.get('in_reply_to_account_id') === status.getIn(['account', 'id']) ? 'reply' : replyIcon} onClick={this.handleReplyClick} counter={status.get('replies_count')} obfuscateCount />
-        <IconButton className={classNames('status__action-bar__button', { reblogPrivate })} disabled={!publicStatus && !reblogPrivate} active={status.get('reblogged')} title={reblogTitle} icon='retweet' onClick={this.handleReblogClick} counter={withCounters ? status.get('reblogs_count') : undefined} />
-        <IconButton className='status__action-bar__button star-icon' animate active={status.get('favourited')} title={intl.formatMessage(messages.favourite)} icon='star' onClick={this.handleFavouriteClick} counter={withCounters ? status.get('favourites_count') : undefined} />
+        <IconButton className={classNames('status__action-bar__button', { reblogPrivate })} disabled={!publicStatus && !reblogPrivate} active={status.get('reblogged')} title={reblogTitle} icon='retweet' onClick={this.handleReblogClick} counter={(withCounters !== false) ? status.get('reblogs_count') : undefined} />
+        <IconButton className='status__action-bar__button star-icon' animate active={status.get('favourited')} title={intl.formatMessage(messages.favourite)} icon='star' onClick={this.handleFavouriteClick} counter={(withCounters !== false) ? status.get('reblogs_count') : undefined} />
```

Comment:
This is a really convoluted way to switch the default, and therefore negate the use of `withCounters` as it's only ever used to enable the counts (on the “Explore” tab), never to explicitly disable them (which is the intended default)

Reply:
> This is another example where the variable is not initialized, so the conditional logic doesn't actually do anything. In this case, the `withCounters` variable is not initialized, so the conditional logic will always evaluate to `false`. This line must be paired with the change I made in status_list.js to yield a working conditional.

`withCounters` is passed from `StatusList`, and `StatusList` has its `withCounters` set to `true` in `app/javascript/mastodon/features/explore/statuses.js`. This is as explained: it is set in the Explore tab only.

> None of these conditionals rely on the application's context/global state. None check the `window` element to see where the user happens to be within the application's navigation tree. If there's some contextual logic that stems from the patchwork of undefined variables, then that logic should be made explicit.

We're not checking on the `window` element, we're passing the prop down where it matters.